cumulus_library/studies/core/builder_condition.py

```diff
@@ -43,38 +19,62 @@ class CoreConditionBuilder(base_table_builder.BaseTableBuilder):
     display_text = "Creating Condition tables..."
 
     def denormalize_codes(self):
-        preferred_config = sql_utils.CodeableConceptConfig(
-            source_table="condition",
-            source_id="id",
-            column_hierarchy=[("code", dict)],
-            target_table="core__condition_codable_concepts_display",
-            filter_priority=True,
-            code_systems=[
-                "http://snomed.info/sct",
-                "http://hl7.org/fhir/sid/icd-10-cm",
-                "http://hl7.org/fhir/sid/icd-9-cm",
-                "http://hl7.org/fhir/sid/icd-9-cm/diagnosis",
-                # EPIC specific systems
-                "urn:oid:1.2.840.114350.1.13.71.2.7.2.728286",
-                "urn:oid:1.2.840.114350.1.13.71.2.7.4.698084.10375",
-                # Spec allowed code of last resort
-                "http://terminology.hl7.org/CodeSystem/data-absent-reason",
-            ],
-        )
-        self.queries.append(
-            base_templates.get_codeable_concept_denormalize_query(preferred_config)
-        )
-
-        all_config = sql_utils.CodeableConceptConfig(
-            source_table="condition",
-            source_id="id",
-            column_hierarchy=[("code", dict)],
-            target_table="core__condition_codable_concepts_all",
-            filter_priority=False,
-        )
-        self.queries.append(
-            base_templates.get_codeable_concept_denormalize_query(all_config)
-        )
+        configs = [
+            sql_utils.CodeableConceptConfig(
+                source_table="condition",
+                column_hierarchy=[("category", list)],
+                target_table="core__condition_dn_category",
+                # This is an extensible binding, and US Core already suggests three
+                # different code systems to pull its recommended four values from.
+                # So let's not filter by system here.
+            ),
```

Comment:
I added three new denormalization tables for condition, because in testing the hypertension upgrade, I noticed my unit test conditions weren't appearing anymore. The old SQL was unnesting directly on the condition table, and so if a row didn't have any entries to unnest, it would be skipped and not make it into core__condition.

So I switched us to separate denormalization here.